### DISCUSSION TITLE
CTBase: make a new release

### DISCUSSION THREAD

**Discussion by ocots** [control-toolbox/control-toolbox.github.io#39] [General]
The idea of this discussion is to point out what we have to finalize a minima to make a new release of `CTBase`.

- [x] [Revise functional](https://github.com/control-toolbox/CTBase.jl/issues/67)
- [x] [REPL](https://github.com/control-toolbox/CTBase.jl/issues/64): stick to the current grammar.
- [ ] [Differential geometry](https://github.com/control-toolbox/CTBase.jl/issues/52): add Poisson bracket of two Hamiltonians and the macro `@Poisson`.
- [x] [Names](https://github.com/control-toolbox/CTBase.jl/issues/40)
- [ ] direct solve using ADNLP 0.6.2  - check [this](https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl/pull/127#issuecomment-1550406816)

Extra:

- [plot!](https://github.com/control-toolbox/CTBase.jl/issues/65)
- [REPL](https://github.com/control-toolbox/CTBase.jl/issues/64): update the grammar.

**Comment by jbcaillau** [comment]
@ocots @BaptisteCbl hi again: well done and THANKS for the `Fixed / NonFixed` stuff. Tests passed, we can move forward 👍🏽👍🏽👍🏽

**Reply by jbcaillau** [reply]
And thanks also for `constraint!` rewriting using more or less one single code (boxes with full info). Nicer and *safer* 👍🏽. Will be reused by the [kwargs version](https://github.com/control-toolbox/CTBase.jl/issues/67#issue-1708927214).

**Comment by jbcaillau** [comment]
@ocots @PierreMartinon @joseph-gergaud bonsoir à tous ; Pierre & Joseph, we need to talk about the todo's for direct solving according to current changes in `CTBase.jl`. This Wednesday 16:00 (or later)?

**Reply by ocots** [reply]
Not possible for me on Wednesday. Tuesday afternoon, Thursday, Friday ?

**Reply by jbcaillau** [reply]
Thursday & Friday are supposed to be off. Tuesday from 17:00 is OK for me.

**Reply by joseph-gergaud** [reply]
17h00 is also ok for me

**Comment by jbcaillau** [comment]
@PierreMartinon @joseph-gergaud sparse Jacobian and Hessian available in ADNLP, to test
https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl/pull/127#issuecomment-1550406816

**Comment by ocots** [comment]
Still need to pursue `Differential Geometry`, to update the doc and to add unit tests.
